tests/test_license.py

```diff
@@ -63,6 +65,20 @@ def test_simple_model_with_annual_licence(simple_linear_model):
     remaining -= remaining / (365 - 2)
     assert_allclose(lic.value(m.timestepper._next, si), remaining / (365 - 3))
 
+def test_annual_license_json(solver):
+    model = load_model("annual_license.json")
+
+    model.timestepper.start = "2001-01-01"
+    model.timestepper.end = "2001-01-31"
+    model.timestepper.delta = 5
```

Comment:
Test is convenient when the time-step fits in to the year exactly. I assume this gets a little messy when that's not the case. Do the existing licence tests cover this situation?

Reply:
I did think about this when I was writing this test. I'm not sure what the behavior is (or even what the correct behavior should be!)